### PR TITLE
feat: Create MultiInputStream API primitive

### PR DESCRIPTION
**Pull request by toubatbrian**
This class could solves the issue where user want the room to stay alive event after participant disconnected and being able to pipe the audio stream from new participants.

Related PR: https://github.com/livekit/agents-js/pull/1018

## Summary by CodeRabbit

* **New Features**
  * MultiInputStream primitive for managing multiple input streams dynamically.
  * Audio input now exposes a close operation to shut down underlying inputs.

* **Bug Fixes / Reliability**
  * Improved handling of audio recorder and subscribed track lifecycle for cleaner shutdown and interception behavior.

* **Tests**
  * Comprehensive tests covering MultiInputStream behavior, concurrency, edge cases, and error handling.

---

<a href="https://livekit.devinenterprise.com/review/livekit/agents-js/pull/1036" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: agents/src/stream/multi_input_stream.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+        const { done, value } = await reader.read();
+        if (done) break;
+
+        // Double-check after the (potentially long) read.
+        if (this.shouldStopPumping(id)) break;
+
+        await this.writer.write(value);
+      }
+    } catch (e) {
```
Comment (lukasIO) [inline agents/src/stream/multi_input_stream.ts:154]: this catch clause silently swallows all kinds of errors, is this intended?

Comment (toubatbrian) [reply agents/src/stream/multi_input_stream.ts:154]: No it's not intended. Made a fix.

**Review by lukasIO — APPROVED** [review]
